src/sorted_dict.jl

```diff
@@ -29,23 +29,93 @@ mutable struct SortedDict{K, D, Ord <: Ordering} <: Associative{K,D}
+a concrete key type, or to use one of the constructors below in
+which the key type is inferred.**
+"""
 SortedDict() = SortedDict{Any,Any,ForwardOrdering}(Forward)
+
+"""
+    SortedDict(o=Forward)
+and
+    SortedDict{K,V}(o=Forward)
+
+Construct an empty `SortedDict` with key type `K` and value type
+`V`. If `K` and `V` are not specified, the dictionary defaults to a
+`SortedDict{Any,Any}`. Keys and values are converted to the given
+type upon insertion. Ordering `o` defaults to `Forward` ordering.
+
+**Note that a key type of `Any` or any other abstract type will lead
+to slow performance, as the values are stored boxed (i.e., as
+pointers), and insertion will require a run-time lookup of the
+appropriate comparison function. It is recommended to always specify
+a concrete key type, or to use one of the constructors below in
+which the key type is inferred.**
+"""
 SortedDict(o::Ord) where {Ord <: Ordering} = SortedDict{Any,Any,Ord}(o)
 
 # Construction from Pairs
 # TODO: fix SortedDict(1=>1, 2=>2.0)
+"""
+    SortedDict(k1=>v1, k2=>v2, ...)
+and
+    SortedDict{K,V}(k1=>v1, k2=>v2, ...)
+
+Construct a `SortedDict` from the given key-value pairs. If `K` and
+`V` are not specified, key type and value type are inferred from the
+given key-value pairs, and ordering is assumed to be `Forward`
+ordering.
+"""
 SortedDict(ps::Pair...) = SortedDict(Forward, ps)
-SortedDict(o::Ordering, ps::Pair...) = SortedDict(o, ps)
 SortedDict{K,D}(ps::Pair...) where {K,D} = SortedDict{K,D,ForwardOrdering}(Forward, ps)
+
+"""
+    SortedDict(o, k1=>v1, k2=>v2, ...)
+
+Construct a `SortedDict` from the given pairs with the specified
+ordering `o`. If `K` and `V` are not specified, the key type and
+value type are inferred from the given pairs. See below for more
+information about ordering.
+"""
+SortedDict(o::Ordering, ps::Pair...) = SortedDict(o, ps)
+
+"""
+    SortedDict{K,V}(o, k1=>v1, k2=>v2, ...)
+
+Construct a `SortedDict` from the given pairs with the specified
+ordering `o`. If `K` and `V` are not specified, the key type and
+value type are inferred from the given pairs. See below for more
+information about ordering.
+"""
 SortedDict{K,D}(o::Ord, ps::Pair...) where {K,D,Ord<:Ordering} = SortedDict{K,D,Ord}(o, ps)
 
 # Construction from Associatives
+
 SortedDict(o::Ord, d::Associative{K,D}) where {K,D,Ord<:Ordering} = SortedDict{K,D,Ord}(o, d)
 
 ## Construction from iteratables of Pairs/Tuples
 
 # Construction specifying Key/Value types
 # e.g., SortedDict{Int,Float64}([1=>1, 2=>2.0])
+"""
+    SortedDict(iter, o=Forward)
+and
+    SortedDict{K,V}(iter, o=Forward)
+
+Construct a `SortedDict` from an arbitrary iterable object of
+`key=>value` pairs. If `K` and `V` are not specified, the key type
+and value type are inferred from the given iterable. The ordering
+object `o` defaults to `Forward`.
+"""
 SortedDict{K,D}(kv) where {K,D} = SortedDict{K,D}(Forward, kv)
 function SortedDict{K,D}(o::Ord, kv) where {K,D,Ord<:Ordering}
```

Comment:
The function signature here doesn't match the one in the comments above.  Might need to rearrange the definitions a little.

Reply:
The problem is is similar to https://github.com/JuliaCollections/DataStructures.jl/issues/340 and should probably be adressed in an other PR